components/tidb_query_expr/src/impl_time.rs

```diff
@@ -329,7 +363,7 @@ pub fn period_diff(p1: Option<&Int>, p2: Option<&Int>) -> Result<Option<Int>> {
 
 #[rpn_fn(nullable, capture = [ctx])]
 #[inline]
-pub fn last_day(ctx: &mut EvalContext, t: Option<&Time>) -> Result<Option<Time>> {
+pub fn last_day(ctx: &mut EvalContext, t: Option<&DateTime>) -> Result<Option<DateTime>> {
```

Comment:
For `None -> Ok(None)`- function, you can remove the `nullable` from `rpn_fn`  attribute, then type `Option<&DateTime>` in args can be `&DateTime`.

Reply:
done

For the existing functions maybe another PR is needed?